File: tools/capture_viewer.py
#!/usr/bin/env python3
"""Screenshot editor/viewer.html via playwright (codespace). Writes /tmp/facility-v3-shot.png."""
import sys
from playwright.sync_api import sync_playwright

url = sys.argv[1] if len(sys.argv) > 1 else "http://127.0.0.1:8123/viewer.html"
out = sys.argv[2] if len(sys.argv) > 2 else "/tmp/facility-v3-shot.png"
with sync_playwright() as p:
    b = p.chromium.launch(args=["--use-gl=angle", "--enable-unsafe-swiftshader"])
    pg = b.new_page(viewport={"width": 1400, "height": 900})
    pg.goto(url)
    pg.wait_for_function("window.__ready === true", timeout=20000)
    pg.wait_for_timeout(1500)  # let first frames settle
    pg.screenshot(path=out)
    b.close()
print("shot-ok", out)

File: tools/capture_viewer.mjs
// Screenshot editor/viewer.html via playwright (swiftshader software GL; resolves from the
// repo's own node_modules, with the historical codespace path as fallback).
// usage: node capture_viewer.mjs [url] [out]
import { createRequire } from 'node:module';
import { existsSync } from 'node:fs';
const local = new URL('../node_modules/', import.meta.url).pathname;
const require = createRequire(existsSync(local) ? local : '/workspaces/vexea-international/node_modules/');
const { chromium } = require('playwright');

const url = process.argv[2] || 'http://127.0.0.1:8123/viewer.html';
const out = process.argv[3] || '/tmp/facility-v3-shot.png';
const b = await chromium.launch({ args: ['--enable-unsafe-swiftshader', '--use-gl=angle', '--use-angle=swiftshader', '--in-process-gpu', '--no-sandbox'] });
const pg = await b.newPage({ viewport: { width: 1400, height: 900 } });
pg.on('console', m => console.log('[pg]', m.type(), m.text()));
pg.on('pageerror', e => console.log('[pgerr]', e.message));
await pg.goto(url);
try { await pg.waitForFunction('window.__ready === true || window.__err', null, { timeout: 15000 }); } catch {}
await pg.waitForTimeout(1500);
await pg.screenshot({ path: out });
await b.close();
console.log('shot-ok', out);
